### PR TITLE
Add feature slot-replayer

### PR DESCRIPTION
**Pull request by bilboquet**
if feature is enabled:
- don't do speculative execution
- don't take time_cursor() into account while initializing block_sequencer
- print ledger changes while replaying slots

### CODE REVIEW

**Review by AurelienFT — APPROVED** [review]

**Review by AurelienFT — DISMISSED** [review]

File: massa-execution-worker/src/slot_sequencer.rs [hunk 1]
```diff
@@ -209,9 +209,16 @@ impl SlotSequencer {
     pub fn update(
         &mut self,
         mut new_consensus_final_blocks: HashMap<Slot, BlockId>,
+        #[cfg_attr(feature = "slot-replayer", allow(unused_assignments))]
         mut new_blockclique: Option<HashMap<Slot, BlockId>>,
         mut new_blocks_metadata: PreHashMap<BlockId, ExecutionBlockMetadata>,
     ) {
+        // as a precaution, if we don't want to process speculative slots, clear the speculative feed just in case
```
Comment (sydhds) [inline massa-execution-worker/src/slot_sequencer.rs:216]: Maybe add a note to specify that in the slot replayer we are only going to process final slot ?

File: massa-execution-worker/src/slot_sequencer.rs [hunk 2]
```diff
@@ -115,21 +115,21 @@ impl SlotSequencer {
...
+                .expect("latest_consensus_final_slots is empty"),
+            initial_blockclique
+                .keys()
+                .max()
+                .copied()
+                .unwrap_or_else(|| Slot::new(self.config.last_start_period, 0)),
         );
 
+        #[cfg(not(feature = "slot-replayer"))]
+        let max_slot = std::cmp::max(max_slot, self.get_time_cursor());
```
Comment (sydhds) [inline massa-execution-worker/src/slot_sequencer.rs:131]: Add a note about why we disable this for the slot replayer?

File: massa-execution-worker/src/slot_sequencer.rs [hunk 3]
```diff
@@ -589,6 +596,7 @@ impl SlotSequencer {
         }
 
         // Check if the next candidate slot is available for execution.
+        #[cfg(not(feature = "slot-replayer"))]
```
Comment (sydhds) [inline massa-execution-worker/src/slot_sequencer.rs:599]: Same as before add a note explaining why we need to disable this for the slot replayer?

File: massa-execution-worker/src/slot_sequencer.rs [hunk 4]
```diff
@@ -706,6 +714,7 @@ impl SlotSequencer {
         // Here we know that there are no SCE-final slots to execute.
 
         // Low priority: execute the next candidate slot that is available for execution, if any.
+        #[cfg(not(feature = "slot-replayer"))]
```
Comment (sydhds) [inline massa-execution-worker/src/slot_sequencer.rs:717]: Same as before :)

File: massa-execution-worker/src/slot_sequencer.rs (hunk 1 above)
Comment (bilboquet) [reply massa-execution-worker/src/slot_sequencer.rs:216]: done

File: massa-execution-worker/src/slot_sequencer.rs (hunk 3 above)
Comment (bilboquet) [reply massa-execution-worker/src/slot_sequencer.rs:599]: done

**Pull request merged**